tools/webdriver/webdriver/bidi/modules/_module.py

```diff
@@ -48,26 +58,33 @@ def __set_name__(self, owner: Any, name: str) -> None:
         params_fn = self.params_fn
         result_fn = self.result_fn
 
-        @functools.wraps(params_fn)
-        async def inner(self: Any, **kwargs: Any) -> Any:
-            params = params_fn(self, **kwargs)
-
-            # Convert the classname and the method name to a bidi command name
-            mod_name = owner.__name__[0].lower() + owner.__name__[1:]
-            if hasattr(owner, "prefix"):
-                mod_name = f"{owner.prefix}:{mod_name}"
-            cmd_name = f"{mod_name}.{to_camelcase(name)}"
-
-            future = await self.session.send_command(cmd_name, params)
-            result = await future
-
-            if result_fn is not None:
-                # Convert the result if we have a conversion function defined
-                result = result_fn(self, result)
-            return result
-
-        # Overwrite the method on the owner class with the wrapper
-        setattr(owner, name, inner)
+        # Create two versions of the command. First will apply `result_fn`
+        # to the result of the command, second with suffix `_raw` will
+        # return the result of the command without modifications.
+        for suffix in ["", "_raw"]:
```

Comment:
In practice for WebDriver tests we have quite a lot of places where we do things like convert unexpected behaviour into exceptions, so if we want to check the actual response we end up writing a wrapper function around the raw command that directly returns the response object. Here's a [random example](https://searchfox.org/mozilla-central/source/testing/web-platform/tests/webdriver/tests/add_cookie/add.py#11-14) of `add_cookie` using this pattern.

Given that precedent I'm inclined to believe that getting the raw protocol response rather than anything post-processed is something we're going to want for testing many commands. On the other hand the raw response is overkill when we are just using a command to setup another test (e.g. navigating, or running some script to set a specific state). So I think there really are two distinct use cases here, and in the existing WebDriver tests not building infrastructure to support them both ended up with all the test modules implementing a bespoke version of raw command execution.

Therefore I think that just exposing `execute_command_raw` (which is basically just the existing `session.send_command`) is going to repeat the same mistakes again. A clear API for sending a well-formed command with appropriate parameter defaults, but just returning the protocol result rather than doing any post-processing seems valuable.

I agree that auto-creating an `_raw` command is surprising. But the metaprogramming is already creating an API that doesn't quite look like the code, and I don't think it would really be that hard to learn. However there are other designs we could consider e.g. making `raw_result` a parameter to all the command methods.

Reply:
`raw_result` parameter sounds like a good option.

Reply:
Alright, changed the approach to use `raw_result` parameter.